packages/web-frontend/src/utils/periodGranularities.js

```diff
@@ -193,9 +203,16 @@ const getDefaultDatesForSingleDateGranularities = (periodGranularity, defaultTim
       singleDateConfig = defaultTimePeriod;
     }
 
+    const validDateOffsetUnit = GRANULARITIES_WITH_ONE_DATE_VALID_OFFSET_UNIT[periodGranularity];
+    if (singleDateConfig.unit !== validDateOffsetUnit) {
+      throw new Error(
```

Comment:
I've checked the production dashboards and map overlays to see if any are configured with an offset unit that doesn't match the periodGranularity, and there are none. This will prevent devs from doing so in the future.